roles/cdi/tasks/provision.yml

```diff
@@ -0,0 +1,26 @@
+---
+# General State of World Checks
+# thought I might need this in case had to do OC specific things
+# but seems I do not, but leaving for now for future use
+- name: Determine Environment
+  shell: "oc version> /dev/null 2>&1; if [ \"$?\" -eq 127 ]; then echo 'kubectl'; else echo 'oc'; fi"
+  register: cli
+
+- name: Check if CDI is already running
+  shell: "kubectl get deployment -n {{ cdi_namespace }} | grep -w 'cdi-deployment' | awk '{ print $1 }'"
+  register: cdi_running
+  ignore_errors: yes
```

Comment:
Not sure here but would kubectl apply -f handle this in one step rather than having to check for the deployment and then conditionally create?  I'm wondering because if CDI setup ever grows beyond a single deployment (ie. clusterroles) we need to handle cases where only part of the deployment is present.